src/encoder/mod.rs

```diff
@@ -652,6 +645,30 @@ impl<'a, W: 'a + Write + Seek, T: ColorType> ImageEncoder<'a, W, T> {
         Ok(())
     }
 
+    /// Write strips from data
+    pub fn write_data(mut self, data: &[T::Inner]) -> TiffResult<()>
+    where
+        [T::Inner]: TiffValue,
+    {
+        let num_pix = usize::try_from(self.width)?.checked_mul(usize::try_from(self.height)?)
+            .ok_or_else(|| ::std::io::Error::new(
+                ::std::io::ErrorKind::InvalidInput,
+                "Image width * height exceeds usize"))?;
+        if data.len() < num_pix {
```

Comment:
But you don't have `data.len()` in `ImageEncoder::new()`. I can put the check in `new` but i will need to store the `num_pix` to compare with `data.len()` on `write_data`

Reply:
But you don't have the data len in the 

> I think this will be a nice improvement, I only have a few simple suggestions for some small changes.
> 
> When you are finished with these you wouldn't mind doing a simple git squash before we merge?

Can't you squash when you merge the PR ? (in gitlab they allow you to make it like a single commit)
But yes i can squash :wink:

Reply:
Haha, that's of course true. My bad